### PR TITLE
docs: complete documentation for Ianvs lifelong learning comprehensive example restoration

### LINKED ISSUE #287: [Bug] NaN Accuracy Metrics in Lifelong Learning Semantic Segmentation Example
## Context

Greetings @hsj576 ,

I'm Abhishek Kumar, LFX mentee working on the Ianvs example restoration project. I need guidance on resolving a critical issue with the lifelong learning semantic segmentation example.

## Problem

The benchmark runs successfully end-to-end and training completes, but all evaluation metrics return **NaN** values instead of numerical accuracy scores.

## What's Working ✅

- Benchmark completes all 6 rounds without crashes
- Training phase succeeds for all rounds
- Model checkpoints generated (181MB files)
- GPU acceleration functional
- No errors or exceptions during execution

## What's Not Working ❌

All evaluation metrics show `nan`:

| Round | Test Samples | Accuracy | Result |
|-------|--------------|----------|--------|
| 0     | 1            | nan      | Expected (insufficient data) |
| 1     | 48           | nan      | Unexpected |
| 2     | 48           | nan      | Unexpected |
| 3     | 48           | nan      | Unexpected |
| 4     | 48           | nan      | Unexpected |
| 5     | 48           | nan      | Unexpected |

## Environment

- **OS**: WSL2 Ubuntu 22.04
- **Python**: 3.10 / 3.12 (tested both)
- **PyTorch**: 2.x with CUDA 11.8
- **GPU**: NVIDIA RTX 4050
- **Ianvs**: Latest from main branch

## Steps to Reproduce
```bash
git clone https://github.com/kubeedge/ianvs.git
cd ianvs
python3 -m venv venv
source venv/bin/activate
pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118
pip install -e .
ianvs -f examples/robot/lifelong_learning_bench/semantic-segmentation/benchmarkingjob-simple.yaml
```

After completion, final results table shows `nan` for all metrics.

## What I've Investigated

- Verified test dataset files exist with valid image paths
- Checked model outputs are being generated
- Examined metrics calculation functions
- Tested on multiple Python versions
- Confirmed training completes successfully with proper checkpoints

Despite investigation, I cannot identify why metrics calculate as NaN.

## Evidence

<details>
<summary>Test Dataset Verification</summary>
```bash
$ wc -l model_eval-*.txt
    1 model_eval-0.txt
   48 model_eval-1.txt
   48 model_eval-2.txt
   48 model_eval-3.txt
   48 model_eval-4.txt
   48 model_eval-5.txt
```
</details>

<details>
<summary>Model Checkpoint Verification</summary>
```bash
$ ls -lh output/train/0/seen_task/global.model
-rw-r--r-- 181M global.model
```
Training produces valid model files.
</details>

## Questions

**I need guidance on:**

1. **Where should I look to debug this?** Which files/functions handle the metrics calculation?
2. **What's the expected format** for predictions and ground truth labels during evaluation?
3. **Is this a known issue** with this example?
4. **What could cause NaN** in the metrics calculation despite successful training?

## Additional Context

During restoration, I fixed multiple compatibility issues to get the benchmark running (empty dataset handling, scheduler fixes, method name corrections, etc.). The pipeline now executes successfully, but the metrics issue remains.

Any guidance on debugging this would be greatly appreciated.

Image
Image

---

**Abhishek Kumar**  
LFX Mentee - KubeEdge Ianvs  
GitHub: @abhishek-8081

### PR DESCRIPTION
**Pull request by abhishek-8081**
## Description

Adds comprehensive documentation for the Ianvs Lifelong Learning Semantic Segmentation Benchmark restoration completed during LFX Term 3. Includes full issue-resolution logs, day-wise implementation notes, and validated benchmark results.

## Summary

* 17/17 issues resolved
* Metrics validated: Accuracy 0.72, mIoU 0.69, fwIoU 0.71
* Multi-round execution enabled

## Fixes

Fixes: #287, #263, #230

## Additional Context

Org: KubeEdge
Mentors: @MooreZheng @hsj576 
Program: LFX Term 3

### CODE REVIEW

**Comment by abhishek-8081** [comment]
Please review this .
@MooreZheng @hsj576

**Review by hsj576 — CHANGES_REQUESTED** [review]
Two minor comments:
1. This document belongs in ianvs/docs/guides.
2. We need to add installation instructions for the sedna package here.

**Comment by hsj576** [comment]
Please squash the commits into a single commit.

**Comment by abhishek-8081** [comment]
@hsj576 I’ve squashed the commits into one. Please review when you get a chance. 
Thanks!

**Comment by abhishek-8081** [comment]
> @hsj576 I’ve squashed the commits into one. Please review when you get a chance. Thanks!

@hsj576 Please Review it once .

**Comment by phantom-712** [comment]
# PR #298 - Detailed Review

**Recommendation: Address Maintainer Feedback and Merge**


#### Introduction

PR #298 is the condensed documentation PR from the LFX Mentorship 2025 Term 3 cluster. It provides a comprehensive guide (2,199 lines) detailing the issue-resolution logs and day-wise implementation notes, acting as the summary of what has already been fixed by this example.

#### Module Details

**Files changed:** 1 file - `docs/guides/documentation_example_restoration.md` (2,199 lines added). Scope: Documentation only. **State:** Open (still active). **Author:** abhishek-8081.

#### Reviewer Coverage & Maintainer Feedback

Maintainer `hsj576` has actively reviewed this PR and provided the following actionable feedback:
1. Confirmed the document correctly belongs in `ianvs/docs/guides`.
2. Requested installation instructions for the `sedna` package.
3. Requested squashing of commits into a single commit.

The PR author has since squashed the commits and requested another review.

#### Consolidated Findings for PR #298

This PR holds significant organizational and pedagogical value as the summary of what was fixed during the Term 3 restoration effort. The day-by-day implementation logs and issue resolution tracking provide critical historical context for the repository.

To bring this PR to a mergeable state, the following should be confirmed:
1. **Sedna Installation Instructions:** The instructions requested by the maintainer (`hsj576`) must explicitly address the Two-Tier Ecosystem divergence (installing from the `resources/third_party/` wheel rather than PyPI) to prevent future contributors from hitting Layer 1 of the crash cascade.
2. **Workflow Alignment:** Align the documentation's "17/17 issues resolved" claim with the finalized merge path of PR #410 and PR #301.

Merging this documentation will provide future contributors with a clear, maintainer-approved guide on how the example was restored and how to set up the environment properly.

**Comment by phantom-712** [comment]
This follows the earlier review at
https://github.com/kubeedge/ianvs/pull/298#issuecomment-4577459158. The earlier
review evaluated the restoration documentation, its historical result claims,
and whether the documented workflow was reproducible from a clean environment.

The current documentation comparison point is Ianvs revision
37a9c60a9747af0cfe3170f84249bc349c56e8d5:
https://github.com/kubeedge/ianvs/commit/37a9c60a9747af0cfe3170f84249bc349c56e8d5

The current runtime uses the vendored Sedna 0.6.0.1 wheel:
https://github.com/kubeedge/ianvs/blob/37a9c60a9747af0cfe3170f84249bc349c56e8d5/resources/third_party/sedna-0.6.0.1-py3-none-any.whl

SHA-256:

```text
16be545309f1f56db388aedb08cd0156525927c1741e96b04bf79243dba54f55
```

The repaired composed Robot working tree completed five lifelong-learning
rounds, evaluation, metric aggregation, knowledge-base interaction, prediction
output, and rank generation. Ianvs returned code 0 and retained 1,641 output
files. The observed accuracy was 0.002232849365268865 for one configured
algorithm.

This result should be recorded in restoration documentation with its source
revision, artifact hash, resource identity, interpreter and platform, run
identifier, output count, and metric semantics. It must not be merged with
older result claims that used a different source revision, dependency stack,
dataset state, or evaluation schedule. A documentation statement such as
"all issues resolved" also needs a table showing which checks were static,
which were bounded, which ran on the exact PR head, and which remain blocked by
external resources.

The current run validates the documentation contract for the repaired composed
tree. It does not prove that PR #298 itself changes the implementation or that
the historical result claims are reproduced by the current environment.
